gaia-component.js

```diff
@@ -122,6 +54,14 @@ var base = {
         this.setInnerDirAttributes = setInnerDirAttributes.bind(null, this);
         document.addEventListener('dirchanged', this.setInnerDirAttributes);
       }
+      if (document.l10n) {
+        this.localize = document.l10n.translateFragment.bind(
+          document.l10n, this.shadowRoot);
+        document.l10n.ready.then(() => {
+          document.addEventListener('DOMRetranslated', this.localize);
+          this.localize();
+        });
+      }
```

Comment:
Not sure `setupShadowL10n` is the correct name perhaps `.localizeShadow(nodes)` which our `.setupShadowL10n()` helper could call internally.